pkg/operator/controller.go

```diff
@@ -120,6 +121,7 @@ func (c *Controller) createOrUpdateResources(cr *imageregistryv1.Config) error {
 
 	err := verifyResource(cr)
 	if err != nil {
+		metrics.Degraded(true)
```

Comment:
It's better to calculate this value based on the degraded condition.

Reply:
I guess that then, if the storage is not configured, we will alert it as Degraded. I guess the goal was to have two distinct alerts: if it is Degrade by any other reason but not Storage Not Configured and it does not have Storage configured for more than X days.

Reply:
Conditions have Reasons, and we can use them to distinguish StorageNotConfigured from other problems.

Reply:
Ah, I see. You mean setting the Degraded counter after we know the reason for the degradation. Let me see if i can achieve this.

Reply:
Done, could you check now?

Reply:
Hey @dmage , i am resolving this comment as I have already implemented the requested changes and you haven't yet provided any other input.